ci/scripts/build.sh

```diff
@@ -48,9 +48,6 @@ cargo build \
 # the file name suffix of artifact for risingwave_java_binding is so only for linux. It is dylib for MacOS
 artifacts=(risingwave sqlsmith compaction-test backup-restore risingwave_regress_test risedev-dev delete-range-test librisingwave_java_binding.so)
 
-echo "--- Compress debug info for artifacts"
```

Comment:
I guess it should always be better to still compress debug info on CI, which can reduce the time usage of artifact uploading/downloading.

Reply:
But according to @zwang28’s test the compressed debuginfo will cause much more memory consumption (400mb to 2G). I have reproduced it as well. Don’t know the reason. 🥹

Reply:
Compression decreases binary size but involves memory overhead. It sounds reasonable? 🤔

Reply:
Agree, but the additional memory usage should not matter on CI since we're mainly caring about the I/O between buildkite steps? Not a big deal. 🥵

Reply:
I guess that won't be a problem. Docker images have its own compression.

Reply:
Oh, this is not the docker image stuff but the ad-hoc building for each CI run on PR or the main branch.

Reply:
Well, I am not familiar with that part. Perhaps we can enable compression in some way?

Reply:
Seems `buildkite artifact upload/download` has no built-in compression. We should still do it ourselves. 🤔